m4/check_java_support.m4

```diff
@@ -83,7 +83,7 @@ AC_DEFUN([AX_CHECK_JAVA_VERSION],
       JNI_LIBDIR=`find $JNI_HOME \( -name "libjvm.so" -or -name "libjvm.dylib" \) \
         | sed "s-/libjvm\.so-/-" \
         | sed "s-/libjvm\.dylib-/-" | head -n 1`
-      JNI_LIBS="-L$JNI_LIBDIR -ljvm"
+      JNI_LIBS="-L$JNI_LIBDIR -ljvm -Wl,-rpath=$JNI_LIBDIR"
```

Comment:
System-wide `ldconfig` requires root and we'd like to avoid that to support
containerized CI.

`LD_LIBRARY_PATH` needs some extra effort in touching some of our other
scripts, and I was hoping that fixing `RPATH`/`RUNPATH` could also spare
some of our users of having to specify LD_LIBRARY_PATH themselves if they
use OS packages. I would strongly recommend that we use RUNPATH if you know
how we can do that individually, though.

On Mon, May 30, 2016 at 9:35 AM, Tibor Benke <notifications@github.com>
wrote:

> In m4/check_java_support.m4
> <https://github.com/balabit/syslog-ng/pull/1070#discussion_r65034497>:
>
> > @@ -83,7 +83,7 @@ AC_DEFUN([AX_CHECK_JAVA_VERSION],
> >        JNI_LIBDIR=`find $JNI_HOME \( -name "libjvm.so" -or -name "libjvm.dylib" \) \
> >          | sed "s-/libjvm\.so-/-" \
> >          | sed "s-/libjvm\.dylib-/-" | head -n 1`
> > -      JNI_LIBS="-L$JNI_LIBDIR -ljvm"
> > +      JNI_LIBS="-L$JNI_LIBDIR -ljvm -Wl,-rpath=$JNI_LIBDIR"
>
> Why ldconfig or LD_LIBRARY_PATH are not enough for a CI system?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/1070/files/a15adc5be5217ab60d7522bb8bd705c996aebc15#r65034497>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AOobpKnDaKX3NCXVIj_z_tnOaBYxikJ6ks5qGpMzgaJpZM4IoZ7j>
> .
>